keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/BaseClusterHandler.kt

```diff
@@ -0,0 +1,51 @@
+package com.netflix.spinnaker.keel.api.plugins
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceDiff
+import com.netflix.spinnaker.keel.api.ResourceSpec
+import com.netflix.spinnaker.keel.diff.toIndividualDiffs
+
+/**
+ * Common cluster functionality.
+ * This class contains abstracted logic where we want all clusters to make the
+ * same decision, or abstracted logic involving coordinating resources or
+ * publishing events.
+ */
+abstract class BaseClusterHandler<SPEC: ResourceSpec, RESOLVED: Any>(
+  resolvers: List<Resolver<*>>
+) : ResolvableResourceHandler<SPEC, Map<String, RESOLVED>>(resolvers) {
```

Comment:
You can probably get rid of `ResolvableResourceHandler` now. The only purpose of that was to carry the reference to `resolvers` for both cluster handlers (which this class serves perfectly now).

Reply:
Well, this is only for clusters. I don't want to remove the resolvable resource handler because it's useful for other resource types I think

Reply:
Fair enough. Was just pointing out that that property used to be in `ResourceHandler` (which was an abstract class before), and I moved it to `ResolvableResourceHandler` when working on the Treasure plugin since it doesn't use resolvers and I needed to convert `ResourceHandler` to an interface. But you're right -- it may be useful some day.